apps/server/src/modules/oauth/controller/api-test/oauth-sso.api.spec.ts

```diff
@@ -315,21 +363,111 @@ describe('OAuth SSO Controller (API)', () => {
 			});
 		});
 
-		describe('when a faulty query is passed', () => {
-			it('should redirect to the login page with an error', async () => {
-				const { user, system } = await setup();
-				currentUser = mapUserToCurrentUser(user);
-				const { state, cookies } = await setupSessionState(system.id);
-				const clientUrl: string = Configuration.get('HOST') as string;
+		describe('when invalid request', () => {
+			it('should throw UnprocessableEntityException', async () => {
+				const { targetSystem, sourceUser, sourceSystem } = await setup();
+				currentUser = mapUserToCurrentUser(sourceUser, undefined, undefined);
+				const { state, cookies } = await setupSessionState(targetSystem.id, true);
 				const query: AuthorizationParams = new AuthorizationParams();
+				query.error = SSOAuthenticationError.INVALID_REQUEST;
+				query.state = state;
+
+				await request(app.getHttpServer()).get(`/sso/oauth/migration`).set('Cookie', cookies).query(query).expect(422);
+			});
+		});
+
+		describe('when invalid request', () => {
+			it('should redirect to the general migration error page', async () => {
+				const { targetSystem, sourceUser, sourceSystem } = await setup();
+				currentUser = mapUserToCurrentUser(sourceUser, undefined, sourceSystem.id);
+				const { state, cookies } = await setupSessionState(targetSystem.id, true);
+				const baseUrl: string = Configuration.get('HOST') as string;
+				const query: AuthorizationParams = new AuthorizationParams();
+				query.error = SSOAuthenticationError.INVALID_REQUEST;
 				query.state = state;
 
 				await request(app.getHttpServer())
 					.get(`/sso/oauth/migration`)
 					.set('Cookie', cookies)
 					.query(query)
 					.expect(302)
-					.expect('Location', `${clientUrl}/login?error=sso_auth_code_step`);
+					.expect(
+						'Location',
+						`${baseUrl}/migration/error?sourceSystem=${sourceSystem.id}&targetSystem=${targetSystem.id}`
+					);
+			});
+		});
+
+		describe('when schoolnumbers mismatch', () => {
+			it('should redirect to the login page with an schoolnumber mismatch error', async () => {
+				const { targetSystem, sourceUser, targetUser, sourceSystem } = await setup();
+				currentUser = mapUserToCurrentUser(sourceUser, undefined, sourceSystem.id);
+				const { state, cookies } = await setupSessionState(targetSystem.id, true);
+				const baseUrl: string = Configuration.get('HOST') as string;
+				const query: AuthorizationParams = new AuthorizationParams();
+				query.code = 'code';
+				query.state = state;
+				jest.setTimeout(999999);
+
+				const idToken: string = jwt.sign(
+					{
+						sub: 'differentExternalUserId',
+						iss: targetSystem.oauthConfig?.issuer,
+						aud: targetSystem.oauthConfig?.clientId,
+						iat: Date.now(),
+						exp: Date.now() + 100000,
+						external_sub: 'differentExternalUserId',
+					},
+					privateKey,
+					{
+						algorithm: 'RS256',
+					}
+				);
+
+				axiosMock
```

Comment:
Can you please move it out from it?
Please also expose the UUIDs that can it be used from outside, or make it possible to pass them.

Can you please find a more specifc naming like axiosMock. Something that explain the context.